### ISSUE TITLE
Remove flex references

### ISSUE THREAD

**Issue by Harbs** [apache/royale-asjs#17]
We need to change reference to reflect the new project name. A quick search in the asjs repo turned up over 10,000 references to flex.

Here's a partial list (what else?)
1. Change asjs package names from `org.apache.flex` to `org.apache.royale`
2. Change compiler package names from `org.apache.flex` to `org.apache.royale`
3. Change environment variable names from `FLEXJS` to `ROYALE`?
4. Documentation annotation from `FlexJS` to `Royale` (i.e. `@productversion FlexJS 0.9` becomes `@productversion Royale 0.9`.)
5. Compiler annotation from flexjs to royale (i.e. `@flexjsignorecoercion`) These should possibly just drop the flexjs part and simply become i.e. `@ignorecoercion`.
6. Are there Flex references in documentation? I assume yes.

**Comment by erikdebruin** [comment]
Is this something we can do in stages?

**Comment by Harbs** [comment]
Probably. I think package paths probably all need to be done at once.
I think that once the packaging branch is merged, we're going to create a new renaming branch to do the renaming. We probably want to get it done ASAP to avoid merge conflicts from the develop branch.
I'm hoping we can split up these tasks to get it done quickly.

**Comment by erikdebruin** [comment]
Count me in for some of the less techie stuff, like documentation...

I think this also brings us to a discussion about project vs. product name, which I'll open on dev@

**Comment by piotrzarzycki21** [comment]
Definitely we need have such discussion. Carlos need to have clear path for the website and information included on it.

**Comment by carlosrovira** [comment]
Hi, I think this should be a branch to work in stages (each commit should normaly be 1 stage) and then tested and merged into develop

**Comment by aharui** [comment]
7. Artifact names
    7a) Maven release coordinates org.apache.flex.*
    7b) IDE packaging apache-flex-*

Also note that "flex" will be used in a few places where we use PixelBender and FlexUnit dependencies. However, I believe I have proven that we don't have references to the main Apache Flex SDK in the packaging branch.

**Comment by aharui** [comment]
source folder names "src/main/flex" -> "src/main/royale"

**Comment by aharui** [comment]
I think we have to leave references to flex package names in flex-compiler-oem and fdb.

**Comment by aharui** [comment]
Also references to FlexToolGroup, FlexRSLInfo and more

**Comment by Harbs** [comment]
Catching up after my holiday. Where do we stand on this?

Is there anything I can do to help? Are we able to merge yet?

**Comment by piotrzarzycki21** [comment]
Hi @Harbs ,

Glad you back! :) Take a look into that conversation [1]. Maybe you could help with FALCON_HOME renaming since you have probably setup ANT build of SDK.

[1] http://apache-royale-development.20373.n8.nabble.com/Maven-Royale-Pipeline-for-feature-rename-branch-td110.html

**Comment by Harbs** [comment]
Question: Should "flex-config" in the *config.xml files be renamed to "royale-config"?

**Comment by piotrzarzycki21** [comment]
I think it may break some things in IDE which supports FlexJS - I would leave it as is for now. We can make this change much later.

**Comment by Harbs** [comment]
Should the target named JSFlex be renamed to Royale? RoyaleJS? Thoughts?

**Comment by Harbs** [comment]
What is flex-tool-api? It seems to be in the maven repo under the flex name? Is that related to flex or royale?

**Comment by piotrzarzycki21** [comment]
https://github.com/apache/royale-asjs/issues/17#issuecomment-336434122,

Nope it's being renamed to "JSRoyale".

**Comment by Harbs** [comment]
OK. It looks like that was not done yet.

**Comment by Harbs** [comment]
Re:
https://github.com/apache/royale-asjs/issues/17#issuecomment-336431433
It looks like Alex already renamed it, but some tests are currently failing due to that rename.

**Comment by Harbs** [comment]
FWIW, there are still 33 instances of `flex-config` in the compiler repo. Should we always expect `royale-config` or should `flex-config` be legal as well so Flex could use the Compiler and use `flex-config` elements?

**Comment by piotrzarzycki21** [comment]
https://github.com/apache/royale-asjs/issues/17#issuecomment-336440894,

Yeah I saw that Alex mention that some tests failing and since it require some deeper refactoring he disabled it, but not sure whether they are the same which you are seeing failing. 

Let's wait for Alex's comments.

**Comment by Harbs** [comment]
FYI, the ant build is working for me locally.

**Comment by aharui** [comment]
I think we want the Royale Compiler to be able to compile regular Flex applications because that might be a step in the migration process.  The Royale Compiler will actually correctly catch some compiler errors that the regular Flex compiler will not catch so proving the code you want to migrate is clean enough to compile with the Royale Compiler makes sense.

So, depending on how easy we want to make it to compile regular Flex apps, we have to leave some names of "flex" things.  One of my considerations is how often folks using Royale for the first time that aren't migrating will see the word "flex".  I think a good goal is to try to get that to zero, but for folks migrating who already know "flex" it is ok for them to see and use "flex" in their configurations.

The compiler should be working off of a default configname.  The default configname was "flex" and I think I already changed that to "royale".  But realize that even in regular Flex, if you were compiling in AIR or AIR Mobile app, you already set configname=air or configname=airmobile and then the compiler goes looking for air-config.xml or airmobile-config.xml.  And we already do that in FlexJS/Royale: you can set configname=js to use js-config.xml for native JS compilation.  However, that means that folks migrating may need to add configname=flex to their configurations.

Similarly, the falcon-flexTasks.jar for Ant got renamed to be falcon-royaleTasks.jar.  Which means that anybody migrating an Ant build will have to edit their build.xml and change that name.  Again, that meets the criteria of having migrating people to some work to use the word "flex" but newbies won't see the word "flex" in their configurations.

If the compiler is assuming flex-config.xml, it might need to change to use configname.

The top-level tag in xxx-config.xml is now <royale-config> instead of <flex-config>, but to avoid requiring migrating folks to edit their existing flex-config.xml files, I tried to get the compiler to allow both <royale-config> and <flex-config> in the xxx-config.xml files.

**Comment by aharui** [comment]
Target names like "JSFlex" were renamed to "JSRoyale".  We can change our minds about that, but my logic is that we have a native cross-compile target of just "JS", then we have fancier targets that factor in all of the FlexJS/Royale framework and that should be called <target platform>Royale.  If we had a WebASM target, that would be "WebASM" for native output and "WebASMRoyale" for any framework we build around that target platform.

Also note that there are flavors of Royale output.  In the Cordova branch there is a JSRoyaleCordova target that packages the output into a Cordova application.  So the formula is for a target name is:

<target-platform><framework name><packaging name>

But we can certainly change that if folks have a better idea.

**Comment by aharui** [comment]
IIRC, flex-tool-api is a release that Chris put in for Maven to try to make it possible to leverage the same look up if the regular Flex compiler were to someday support Maven.  Changing that to royale-tool-api would require a new release of the flex-tool-api.  I just left it alone for now.  I'm not quite sure we really need it to be a separate thing if there aren't any realistic plans to support Maven from Apache Flex.

**Comment by bigosmallm** [comment]
It looks like this ticket can be closed now?

**Comment by piotrzarzycki21** [comment]
@Harbs can you close that one ?
